### PR TITLE
zsh: support extra settings in oh-my-zsh plugins

### PR DESCRIPTION
**Pull request by carlostome**
Hi,

I've added support to pass settings to plugins.

My use case is adding indentities to ssh-agent ([oh-my-zsh/ssh-agent](https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/ssh-agent)).

If there is a more direct way to achieve so, I'll be glad to hear.

### CODE REVIEW

File: modules/programs/zsh.nix [hunk 1]
```diff
@@ -152,6 +152,17 @@ let
           Name of the theme to be used by oh-my-zsh.
         '';
       };
+
+      extraSettings = mkOption {
```
Comment (rycee) [inline modules/programs/zsh.nix]: The standard name for textual extra configuration is `extraConfig` so can here change
```suggestion
      extraConfig = mkOption {
```

File: modules/programs/zsh.nix [hunk 2]
```diff
@@ -152,6 +152,17 @@ let
...
+        default = "";
+        example = ''
+          zstyle :omz:plugins:ssh-agent identities id_rsa id_rsa2 id_github
+        '';
+        type = types.lines;
+        description = ''
+          Extra settings for plugins
```
Comment (rycee) [inline modules/programs/zsh.nix]:
```suggestion
          Extra settings for plugins.
```

**Comment by carlostome** [comment]
I applied your suggestions, you can squash and merge. Many thanks!

File: modules/programs/zsh.nix [hunk 3]
```diff
@@ -398,6 +409,8 @@ in
         ${envVarsStr}
 
         ${optionalString cfg.oh-my-zsh.enable ''
+            # oh-my-zsh extra settings for plugins
+            ${cfg.oh-my-zsh.extraSettings}
```
Comment (dermetfan) [inline modules/programs/zsh.nix]: Seems you forgot to change it to `extraConfig` here

**Pull request merged**